metricbeat/module/php_fpm/php_fpm.go

```diff
@@ -0,0 +1,103 @@
+package php_fpm
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/status"
+)
+
+// HostParser is used for parsing the configured php-fpm hosts.
+var HostParser = parse.URLHostParserBuilder{
+	DefaultScheme: defaultScheme,
+	DefaultPath:   defaultPath,
+}.Build()
+
+// StatsClient provides access to php-fpm stats api
+type StatsClient struct {
+	address  string
+	user     string
+	password string
+	http     *http.Client
+}
+
+// NewStatsClient creates a new StatsClient
+func NewStatsClient(m mb.BaseMetricSet, isFullStats bool) *StatsClient {
+	var address string
+	address = m.HostData().SanitizedURI + "?json"
+	if isFullStats {
+		address += "&full"
+	}
+	return &StatsClient{
+		address:  address,
+		user:     m.HostData().User,
+		password: m.HostData().Password,
+		http:     &http.Client{Timeout: m.Module().Config().Timeout},
+	}
+}
+
+// Fetch php-fpm stats
+func (c *StatsClient) Fetch() (io.ReadCloser, error) {
+	req, err := http.NewRequest("GET", c.address, nil)
```

Comment:
Actually, I was thinking of skipping the http proxy and directly access php-fpm using a fastcgi client. I could find this one [go-fastcgi-client](https://github.com/flashmob/go-fastcgi-client) (since golang's fcgi package is for server only). What do you think?